pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java

```diff
@@ -2580,7 +2581,14 @@ public String getBookkeeperMetadataStoreUrl() {
         } else {
             // Fallback to same metadata service used by broker, adding the "metadata-store" to specify the BK
             // metadata adapter
-            return "metadata-store:" + getMetadataStoreUrl();
+            String suffix;
+            if (StringUtils.isNotBlank(metadataStoreUrl)) {
+                suffix = metadataStoreUrl;
+            } else {
+                // Fallback to old setting
+                suffix = "zk:" + zookeeperServers + new ClientConfiguration().getZkLedgersRootPath();
```

Comment:
should we use some utility method in ClientConfiguration ?
or build a dummy ClientConfiguration and use BK functions to compute the value ?